app/services/organization_update_service.rb

```diff
@@ -15,7 +15,18 @@ def update(organization, params)
       if params.has_key?("partner_form_fields")
         params["partner_form_fields"].delete_if { |field| field == "" }
       end
+
+      if Flipper.enabled?(:enable_packs) && params[:request_unit_names]
+        # Find or create units for the organization
+        request_unit_ids = params[:request_unit_names].reject(&:blank?).map do |request_unit_name|
+          Unit.find_or_create_by(organization: organization, name: request_unit_name).id
+        end
+        params.delete(:request_unit_names)
```

Comment:
OK -- I switched to a duplicate and avoided mutating the original. I used `dup` instead of `deep_dup` so that it would maintain the params require/permit logic. I think.